CryptoPrices/Sources/Domain/Sources/Entities/CoinDetail/MarketData.swift

```diff
@@ -0,0 +1,18 @@
+//
+//  MarketData.swift
+//
+//  Created by Khanh on 28/12/2022.
+//
+
+// sourcery: AutoMockable
+public protocol MarketData {
+
+    var currentPrice: USDDecimalable { get }
+    var priceChangePercentage24H: Double { get }
+    var marketCap: USDDecimalable { get }
+    var marketCapChangePercentage24H: Double { get }
+    var ath: USDDecimalable { get }
+    var athChangePercentage: USDDoublable { get }
+    var atl: USDDecimalable { get }
+    var atlChangePercentage: USDDoublable { get }
+}
```

Comment:
Feel free to remove `// sourcery: AutoMockable` 🙏. Because it just has some getters, and it doesn't have any methods. What we need here is just a quick way to get a dummy instance without putting too much effort dealing with a lot of parameters in the initializer, which you've done that already:
```swift
public extension MockCoinDetail {

     static var single = MockCoinDetail(
         id: "mockID",
         symbol: "mockSymbol",
         name: "MockCoin",
         image: MockImage(large: "https://assets.coingecko.com/coins/images/1/large/bitcoin.png?1547033579"),
         marketData: MockMarketData(
             currentPrice: MockUSDDecimal(usd: 12.34),
             priceChangePercentage24H: 2.3,
             marketCap: MockUSDDecimal(usd: 567.89),
             marketCapChangePercentage24H: -2.1,
             ath: MockUSDDecimal(usd: 123_567.89),
             athChangePercentage: MockUSDDouble(usd: 1_000.0),
             atl: MockUSDDecimal(usd: 1.0),
             atlChangePercentage: MockUSDDouble(usd: 0.01)
         )
     )
 }
```
You can make the same with `Image`, `MarketData`, ...

PS: There were actually no generated codes for this since to do that adding `// sourcery: AutoMockable` alone is not enough. You also have to add the `SourceryPlugin` dependency to the `Entities` module, like what we did with the `RepositoryProtocol` module here:

<img width="926" alt="Screenshot 2022-12-29 at 16 37 10" src="https://user-images.githubusercontent.com/17972674/209932663-e79f22b2-ce48-4e91-af06-8ef999f610e2.png">

Reply:
Oops, this comment should've been placed on the `MockCoinDetail.swift` instead of `MockMarketData.swift` 😅

Reply:
Good catch 🙌 . I fixed it at https://github.com/nimblehq/nimble-crypto-ios/pull/78/commits/6475648e8cd2a72d48a67bebb10810c5f2e8474f.